### PR TITLE
Display an error when a junk URL is entered, and provide suggestions.

### LINKED ISSUE #178: Error message for non-spec-compliant links
https://matrix.to/#/obviousgarbage produces the same output as https://matrix.to

We might want to show an error message if the link is not valid syntax, instead of just a blank screen.

### LINKED ISSUE #179: invalid matrix.to link recognized as valid but fails on mobile
**Describe the bug**
https://matrix.to/#cctg:bubu1.eu is recognized as a valid matrix.to link but it's invalid according to the spec. This leads to the link seeming to work on desktop/in the browser but then failing to work on mobile clients.

**To Reproduce**
1. open https://matrix.to/#cctg:bubu1.eu
2. try to open it on mobile
3. ... it won't load

**Expected behavior**
Don't display the #cctg:bubu1.eu room when visiting https://matrix.to/#cctg:bubu1.eu

**Screenshots**

image

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Archlinux
 - Browser [e.g. chrome, safari] FF 84
 - Version [e.g. 22] matrix.to

**Smartphone (please complete the following information):**
 - Android 8.1
 - Browser [e.g. stock browser, safari] FF 83.1
 - Element Android 1.0.11

### PR DESCRIPTION
**Pull request by DanilaFe**
Should fix #178 and #179. Here are some screenshots, because why not?

When visiting `#/garbage` or `#garbage`:
Screenshot 2021-08-30 at 14-43-26 You're invited to talk on Matrix

When visiting `#cctg:bubu1.eu` (as described in #179)
Screenshot 2021-08-30 at 14-43-11 You're invited to talk on Matrix

### CODE REVIEW

File: src/InvalidUrlView.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                'The link you have entered is not valid. If you like, you can ',
+                t.a({ href: "#/" }, 'return to the home page.')
+            ]),
+            vm.validFixes.length ? this._renderValidFixes(t, vm.validFixes) : [],
+        ]);
+    }
+
+    _describeLinkKind(kind) {
+        switch (kind) {
+            case LinkKind.Room: return "The room ";
```
Comment (bwindels) [inline src/InvalidUrlView.js:34]: Perhaps look at `link.identifierKind`to differentiate between room and room alias here? Wdyt?

**Review by bwindels — COMMENTED** [review]
Excellent! Love the fix suggestions. Just one minor suggestion.

File: src/InvalidUrlView.js (hunk 1 above)
Comment (DanilaFe) [reply src/InvalidUrlView.js:34]: Only thing is, I'm not sure how to make the distinction in "plain terms". Do we want to ask users about _room aliases_? If I click on an invite to `#thisweekinmatrix` and I'm told it's an alias, that seems a little weird.

Comment (bwindels) [reply src/InvalidUrlView.js:34]: I think it's more confusing to give two options labeled "The room". Perhaps label a room id "The room" and a room alias "The room alias"?

Comment (DanilaFe) [reply src/InvalidUrlView.js]: Implemented this in 3802829 :)

**Pull request merged**